debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java

```diff
@@ -216,6 +220,26 @@ private void delaySnapshotIfNeeded(ChangeEventSourceContext context) throws Inte
      */
     protected void connectionCreated(SnapshotContext snapshotContext) throws Exception {
     }
+    
+    private Optional<TableId> toTableId(Set<TableId> tableIds, Pattern id) {
+        return tableIds
+                .stream()
+                .filter(tid -> id.asPredicate().test(tid.toString()))
+                .findFirst();
+    }
+
+    private Set<TableId> sort(Set<TableId> capturedTables) throws Exception {
+        String value = connectorConfig.getConfig().getString(RelationalDatabaseConnectorConfig.TABLE_WHITELIST);
+        if (value != null) {
+            return Strings.setOfRegex(value, Pattern.CASE_INSENSITIVE)
```

Comment:
Could you please take a look at Strings.listOfRegex() method? IMHO you want to keep the order of regexes